### ISSUE TITLE
Using default slot in wrapper components leads to element duplication

### ISSUE THREAD

**Issue by ftroitsky** [withastro/astro#11977]
### Astro Info

```block
Astro                    v4.15.4
Node                     v20.17.0
System                   macOS (x64)
Package Manager          yarn
Output                   static
Adapter                  none
Integrations             @astrojs/react
                         @astrojs/tailwind
```

### Describe the Bug

Following setup:
Layout.astro
```
---
import Header from './Header.astro'
import Footer from './Footer.astro'
---
<Header />
  <slot />
<Footer />
```
index.astro
``` 
---
import Layout from '.../layouts/Layout.astro'
---
<Layout title="Title" description="description">
	<Home />
<Layout />
```

Produces page where Header and footer components are rendered twice in the DOM:

- Header
- Home
- Header
- Footer
- Footer

However if named slots are used everything is fine.
so something like this would work correctly:

```
<Header />
  <slot name="children"/>
<Footer />
---
<Layout title="Title" description="description">
	<Home slot="children"/>
<Layout />
```

### What's the expected result?

The output to the DOM tree should be the following:

- Header
- Home
- Footer

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-z63fyt

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by ArmandPhilippot** [comment]
It seems to me you have a typo and I think this is why your experiencing this issue.

**TL;DR: Use `</Layout>` and not `<Layout />` to close your component!** 😉 

In `src/pages/index.astro`, you're misusing your Layout component:

```astro
<body>
  <Layout>
  <h1>Astro</h1>
  <Layout />
</body>
```

With this syntax, your Layout component is used twice in the page:
* `<Layout>`: a Layout component with children (`h1` **and `<Layout />`**) but without closing tag
* `<Layout />`: a self-closing Layout component without children

This is why you're page is rendered like that:

```
Header // <Layout> begins rendering
H1 // Your children is rendered
Header // <Layout /> (also your children) begins rendering
Footer // <Layout />  (also your children) stops rendering
Footer // <Layout> stops rendering
```

You're body should look like:

```astro
<body>
  <Layout>
    <h1>Astro</h1>
  </Layout>
</body>
```

**Why it works when you use a named slot:**

Your second Layout (`<Layout />`) is once again considered as children. However, since you're not using the slot attribute (`<Layout slot="children" />`), the component is ignored because `<Layout>` does not have a default slot.

**Comment by ftroitsky** [comment]
@ArmandPhilippot OMFG, i really missed this several times. Over-reliance on tooling makes oneself dumb.

Thank you!
